metagraph/src/common/serialization.cpp

```
@@ -350,10 +539,11 @@ bool load_number_string_map(std::istream &in, Map *map) {
         in.seekg(pos, in.beg);
 
         try {
-            size_t const num = NumberSerialisation::deserialiseNumber(in);
+            size_t const num = load_number(in);
             for (size_t i = 0; i < num; ++i) {
-                auto key = NumberSerialisation::deserialiseNumber32(in);
-                std::string value = StringSerialisation::deserialiseString(in);
+                auto key = load_number32(in);
+                std::string value;
+                load_string(in, &value);
```

check the return value or make it throw exceptions instead of returning `bool`

---

same in other similar cases...